cli/src/cmd/snapshot.rs

```diff
@@ -0,0 +1,274 @@
+//! Snapshot command
+
+use crate::cmd::{
+    test,
+    test::{Test, TestOutcome},
+    Cmd,
+};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::{
+    collections::HashMap,
+    fmt::Write,
+    fs,
+    io::{self, BufRead},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use structopt::StructOpt;
+
+/// A regex that matches a basic snapshot entry like
+/// `testDeposit() (gas: 58804)`
+pub static RE_BASIC_SNAPSHOT_ENTRY: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"(?P<sig>(\w+)\s*\((.*?)\))\s*\((gas:)?\s*(?P<gas>\d+)\)").unwrap());
+
+#[derive(Debug, Clone, StructOpt)]
+pub struct SnapshotArgs {
+    /// All test arguments are supported
+    #[structopt(flatten)]
+    test: test::TestArgs,
+    /// Additional configs for test results
+    #[structopt(flatten)]
+    config: SnapshotConfig,
+    #[structopt(
+        help = "Compare against a snapshot and display changes from the snapshot. Takes an optional snapshot file, [default: .gas-snapshot]",
+        long
+    )]
+    diff: Option<Option<PathBuf>>,
```

Comment:
Is this double option needed?

Reply:
yeah so this allows us to pass `--diff` as flag, in which case we take the `.gas-snapshot` file for comparison or

--diff <file> to compare against a specific snapshot file